load-testing/tests/relays_stress_helpers_test.go

```diff
@@ -0,0 +1,1143 @@
+// remaining test duration in blocks, the relay rate per application, the relay
+// cost, and the block duration.
+func (s *relaysSuite) getAppFundingAmount(currentBlockHeight int64) sdk.Coin {
+	currentTestDuration := s.startBlockHeight + s.testDurationBlocks - currentBlockHeight
+	// Multiply by 2 to make sure the application does not run out of funds.
+	appFundingAmount := s.relayRatePerApp * s.relayCost * currentTestDuration * blockDuration * 2
+	return sdk.NewCoin("upokt", math.NewInt(appFundingAmount))
+}
+
+// addPendingFundApplicationMsg generates a bank.MsgSend message to fund a given
+// application and appends it to the funding account's pending messages.
+// No transaction is sent to give flexibility to the caller to group multiple
+// messages in a single transaction.
+func (s *relaysSuite) addPendingFundApplicationMsg(application *accountInfo) {
+	s.addPendingFundMsg(application.accAddress, sdk.NewCoins(application.amountToStake))
+}
+
+// addPendingStakeApplicationMsg generates a MsgStakeApplication message to stake a given
+// application then appends it to the application account's pending messages.
+// No transaction is sent to give flexibility to the caller to group multiple
+// application messages into a single transaction which is useful for staking
+// then delegating to multiple gateways in the same transaction.
+func (s *relaysSuite) addPendingStakeApplicationMsg(application *accountInfo) {
+	application.addPendingMsg(apptypes.NewMsgStakeApplication(
+		application.accAddress.String(),
+		application.amountToStake,
+		[]*sharedtypes.ApplicationServiceConfig{{Service: usedService}},
+	))
+}
+
+// addPendingDelegateToGatewayMsg generates a MsgDelegateToGateway message to delegate
+// a given application to a given gateway then appends it to the application account's
+// pending messages.
+func (s *relaysSuite) addPendingDelegateToGatewayMsg(application, gateway *accountInfo) {
+	application.addPendingMsg(apptypes.NewMsgDelegateToGateway(
+		application.accAddress.String(),
+		gateway.accAddress.String(),
+	))
+}
+
+// sendStakeAndDelegateAppsTxs stakes the new applications and delegates them to both
+// the active and new gateways.
+// It also ensures that new gateways are delegated to the existing applications.
+func (s *relaysSuite) sendStakeAndDelegateAppsTxs(
+	sessionInfo *sessionInfoNotif,
+	newApps, newGateways []*accountInfo,
+) {
+
+	// TODO_IN_THIS_COMMIT: send an UpdateParams message to the application
+	// module to set `max_delegated_gateways` accordingly.
+
+	logger.Debug().
+		Int64("session_num", sessionInfo.sessionNumber).
+		Int64("block_height", sessionInfo.blockHeight).
+		Msgf(
+			"delegating apps for next session %d",
+			sessionInfo.sessionNumber+1,
+		)
+
+	for _, app := range s.activeApplications {
+		for _, gateway := range newGateways {
+			s.addPendingDelegateToGatewayMsg(app, gateway)
+		}
+		s.sendPendingMsgsTx(sessionInfo.blockHeight, app)
+	}
+
+	for _, app := range newApps {
```

Comment:
1. In the next pipeline stage/operation, the `channel.ForEach()` which consumes `stakingSuppliersAndGatewaysObs` and calls `mapStakingInfoWhenStakingAndDelegatingNewApps()`.
2. Unfortunately not because of the stake msg that needs to be included inside of the apps loop that isn't present in `sendDelegateInitialAppsTxs`. I think this method is what any alternative refactoring would reduce to.